src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/AbstractDocumentSpanEntry.cs

```diff
@@ -2,69 +2,73 @@
 
 using System;
 using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Windows;
 using System.Windows.Documents;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Shell.TableControl;
 using Microsoft.VisualStudio.Shell.TableManager;
+using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.FindUsages
 {
     internal partial class StreamingFindUsagesPresenter
     {
         /// <summary>
         /// Base type of all <see cref="Entry"/>s that represent some source location in 
-        /// a <see cref="Document"/>.  Navigation to that location is provided by this type.
+        /// a <see cref="CodeAnalysis.Document"/>.  Navigation to that location is provided by this type.
         /// Subclasses can be used to provide customized line text to display in the entry.
         /// </summary>
         private abstract class AbstractDocumentSpanEntry : Entry
         {
             private readonly AbstractTableDataSourceFindUsagesContext _context;
 
-            private readonly DocumentSpan _documentSpan;
             private readonly string _projectName;
             private readonly object _boxedProjectGuid;
-            protected readonly SourceText _sourceText;
+
+            private readonly SourceText _lineText;
+            private readonly MappedSpanResult _mappedSpanResult;
```

Comment:
MappedSpanResult is a super weird name.

Reply:
It feels like getting rid of Document is fine.  However, it feels super weird that we're not representing a span as a TextSpan, and have this MappedSpan guy.  Is tehre a reason for that?

Reply:
you can suggest better name if you have better name in mind. mappedSpanResult contains a span in it.

Reply:
"MappedSpan".